go/enclave/enclave.go

```diff
@@ -238,7 +241,14 @@ func (e *enclaveImpl) SubmitL1Block(block types.Block, receipts types.Receipts,
 	}
 	e.logger.Info("ProcessL1Block successful", log.BlockHeightKey, block.Number(), log.BlockHashKey, block.Hash())
 
+	_, err = e.rollupManager.ProcessL1Block(&block)
```

Comment:
I'm not sure this sequence is right.
It might work for now, but it highlights that it needs more work to get the abstractions right.

Let's reason through what is needed from a high level when a new L1 block comes in.

I. When we are in catchup mode:
  1. Look for a rollup.
  2. If there is a rollup, then convert it to batches, and execute the batches to build up the l2 chain.
  3. Look for xchain messages, convert them to synthetic txs and execute them as part of one of those batches.

II. We are in live mode:
  1. Look for a rollup.
  2. If there is one, confirm that it matches the l2 chain you are already building live.
  3. Look for xchain messages...